Highlight svg elements

amoose136 opened the pull request:
Adds an import statement to index.php that imports a new script, `highlightPathsInSvgPreview.js` that will make it so that every color in the svg preview is highlighted when the corresponding pass in the passes table is hovered over. The script also removes the fill of any object and sets the outline width of all entities to 1px (hairline). There was also a tweak to the css defined in app.css at the end that highlights the row that is hovered over.

jondale commented:
Since these are plugin specific changes they should live in the specific plugin directory instead of changing the core app.

Specifically,
* move highlightPathsInSvgPreview.js to /cncfm/plugins/uploaders/svg2laser/js/
* put css changes to /cncfm/plugins/uploadesr/svg2laser/css/plugin.css

The core will automatically load those files when appropriate so those are the only 2 files that should change for this pull request.

amoose136 commented:
Javascript and CSS relocated to the indicated locations.
 
Prettier moved some stuff around spacing wise in `svg2laser.js` but the only real change is adding `processSVG();` at the end of the `readsvg` function. (line 327).  The changes to `common.php` and `app.css` should be safe to reject as those are almost all Prettier made changes or things to get it working on my Mac.

- I corrected some problems with certain svg files from FreeCad that were loose with going between the `style="opacity: 1.0"` and the `opacity="1.0"` prop syntax which caused some problems with the highlighting. 
- The script now dims images as well
- The viewport is now adjusted to closely match the bounding box of the svg elements instead of being fixed to the page which you can't even see.
-  Fixed a CSS highlighting inconsistancy for Safari where it rendered differently than Chrome and Firefox.

jondale commented:
I think ultimately we'll want to roll in the changes to the primary class in svg2laser.js but I like having it separate js function now to digest the changes and will merge it like that.

I would like to remove the ini_set calls.  If they are only needed in certain platforms or situations then those settings should go in the systems php.ini file instead.

That change and removing common.php and app.css from the pull request and I think we can merge it.

amoose136 commented:
Still adds `.DS_Store` to `.gitignore` but that should do nothing on other platforms.
